### PR TITLE
Maya: Redshift Volume Loader Implement update, remove, switch + fix vdb sequence support

### PR DESCRIPTION
**Pull request by BigRoy**
## Brief description

Improves Redshift Volume VDB loader to actually support updating and removing + also support single file or multiple frames in a sequence

## Additional info

The Loader still loads the volume **without!** a material assigned by default. I was thinking of potentially auto-assigning a simple Volume Material but wasn't sure whether we wanted that in the same namespace + actually auto-remove that material when removing the volume as well.

For now you'll need to just manually assign the shader you'll want to render it with after loading the volume.

## Testing notes:
1. Load a VDB volume into Maya

---

Tested with Maya 2022 + Redshift 3.0.67

### CODE REVIEW

**Comment by m-u-r-p-h-y** [comment]
I'm probably doing something wrong but it seems like path resolution is not working in my case.

- published vdb sequence from houdini
- loaded vdb to Maya

```
>>> Finding presets for maya:load ...
// openpype.pipeline.load.utils : Running 'LoadVDBtoRedShift' on 'shVDB'
// Warning:  [Redshift] Failed to find volume file D:\PROJECTS\murphy_22101\shots\shVDB\publish\vdbcache\vdbcacheDefault\v001\22101_shVDB_vdbcacheDefault_v001.1001.vdb\22101_shVDB_vdbcacheDefault_v001.0001.vdb. Volume will be empty // 
// Warning:  [Redshift] Failed to find volume file D:\PROJECTS\murphy_22101\shots\shVDB\publish\vdbcache\vdbcacheDefault\v001\22101_shVDB_vdbcacheDefault_v001.1001.vdb\22101_shVDB_vdbcacheDefault_v001.1001.vdb. Volume will be empty // 
Good bye
```

published vdb sequence is 
`D:\PROJECTS\murphy_22101\shots\shVDB\publish\vdbcache\vdbcacheDefault\v001\22101_shVDB_vdbcacheDefault_v001.####.vdb`

**Review by m-u-r-p-h-y — CHANGES_REQUESTED** [review]
I'm probably doing something wrong but it seems like path resolution is not working in my case.

- published vdb sequence from houdini
- loaded vdb to Maya

```
>>> Finding presets for maya:load ...
// openpype.pipeline.load.utils : Running 'LoadVDBtoRedShift' on 'shVDB'
// Warning:  [Redshift] Failed to find volume file D:\PROJECTS\murphy_22101\shots\shVDB\publish\vdbcache\vdbcacheDefault\v001\22101_shVDB_vdbcacheDefault_v001.1001.vdb\22101_shVDB_vdbcacheDefault_v001.0001.vdb. Volume will be empty // 
// Warning:  [Redshift] Failed to find volume file D:\PROJECTS\murphy_22101\shots\shVDB\publish\vdbcache\vdbcacheDefault\v001\22101_shVDB_vdbcacheDefault_v001.1001.vdb\22101_shVDB_vdbcacheDefault_v001.1001.vdb. Volume will be empty // 
Good bye
```

published vdb sequence is 
`D:\PROJECTS\murphy_22101\shots\shVDB\publish\vdbcache\vdbcacheDefault\v001\22101_shVDB_vdbcacheDefault_v001.####.vdb`

File: openpype/hosts/maya/plugins/load/load_vdb_to_redshift.py [hunk 1]
```diff
@@ -87,3 +95,70 @@ def load(self, context, name=None, namespace=None, data=None):
...
+            filename = path
+        else:
+            # The path points to the publish .vdb sequence filepath so we
+            # find the first file in there that ends with .vdb
+            files = sorted(files)
+            first = next((x for x in files if x.endswith(".vdb")), None)
+            if first is None:
+                raise RuntimeError("Couldn't find first .vdb file of "
+                                   "sequence in: %s" % path)
+            filename = os.path.join(path, first)
```
Comment (antirotor) [inline openpype/hosts/maya/plugins/load/load_vdb_to_redshift.py:121]: I think this should fix the issue?

```suggestion
            filename = os.path.join(folder, first)
```

**Review by antirotor — CHANGES_REQUESTED** [review]
It happened to me too,

**Review by m-u-r-p-h-y — APPROVED** [review]
managing vdb caches works as expected.

- loading
- up versioning
- removing

Maya2023 Redshift 3.5.04

image

could not test rendering as we ran out of trial :(

**Review by antirotor — APPROVED** [review]
I've just tested it. The volume loads correctly. Strange thing is that render doesn't match the points in the viewport - it seems like the rendered volume is scaled up, but that might be just my test data - as I can't see anything realted to scale in the code.

**Pull request merged**